runtime/near-vm-errors/src/lib.rs

```diff
@@ -25,6 +25,9 @@ pub enum VMRunnerError {
     WasmUnknownError {
         debug_message: String,
     },
+    UnsupportedCompiler {
```

Comment:
While we are at it, add comment explaining when we emit this error

Reply:
Scratch that, I think I'd just kill this variant

Reply:
too late now^^ but yeah if we delete the precompile code next, then this can be deleted as well

Reply:
https://github.com/near/nearcore/pull/7851

:weary: one does never "just" do a thing